modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java

```diff
@@ -2154,12 +2154,12 @@ private void performDrag(MouseEvent event) {
         if (dragDelta > 0) {
             // Dragging the tab header towards higher indexed tab headers inside headersRegion.
             dragDirection = MIN_TO_MAX;
-        } else {
+        } else if (dragDelta < 0) {
             // Dragging the tab header towards lower indexed tab headers inside headersRegion.
             dragDirection = MAX_TO_MIN;
         }
         // Stop dropHeaderAnim if direction of drag is changed
-        if (prevDragDirection != dragDirection) {
+        if (dragDirection != 0 && prevDragDirection != dragDirection) {
```

Comment:
Did you mean `dragDelta != 0`?

Reply:
it is sort of equivalent in this case...

Reply:
It is the same as using `dragDelta != 0`, since `dragDelta = 0` implies `dragDirection = 0`.

The if-else expression doesn't provide a value for the case `dragDelta == 0`, so I've added an initial value 0 to `dragDirection`. 

Since `prevDragDirection` initial value is -1, we need to verify that `dragDirection` is not zero and also  `prevDragDirection` is different to `dragDirection` to detect a change of direction.

Reply:
Right. Thanks for the clarification.